### PR TITLE
Silence some type conversion warnings and fix a possible bug.

### PR DESCRIPTION
**Pull request by ace20022**
Please have a special look at the commits prefixed [RFC].
2a35535 might not fix a real bug, since the cast for npos might wrap to a negative integer.

### CODE REVIEW

File: xbmc/Util.cpp [hunk 1]
```diff
@@ -342,8 +342,8 @@ std::string CUtil::GetTitleFromPath(const CURL& url, bool bIsFolder /* = false *
   else if (url.IsProtocol("shout"))
   {
     const std::string strFileNameAndPath = url.Get();
-    const int genre = strFileNameAndPath.find_first_of('=');
-    if(genre <0)
+    size_t genre = strFileNameAndPath.find_first_of('=');
```
Comment (Rechi) [inline xbmc/Util.cpp]: why remove const?

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp [hunk 2]
```diff
@@ -1806,7 +1806,7 @@ bool CDVDDemuxFFmpeg::SeekChapter(int chapter, double* startpts)
 
     if(startpts)
     {
-      *startpts = DVD_SEC_TO_TIME(ich->GetChapterPos(chapter));
+      *startpts = DVD_SEC_TO_TIME((double) ich->GetChapterPos(chapter));
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: C++ style cast

File: xbmc/guilib/FFmpegImage.cpp [hunk 3]
```diff
@@ -97,7 +97,7 @@ static int mem_file_read(void *h, uint8_t* buf, int size)
   size_t tocopy = std::min((size_t)size, (size_t)unread);
   memcpy(buf, mbuf->data + mbuf->pos, tocopy);
   mbuf->pos += tocopy;
-  return tocopy;
+  return (int) tocopy;
```
Comment (Rechi) [inline xbmc/guilib/FFmpegImage.cpp]: C++ style cast

File: xbmc/guilib/GUIWindowManager.cpp [hunk 4]
```diff
@@ -1463,7 +1463,7 @@ void CGUIWindowManager::DispatchThreadMessages()
 
   CSingleLock lock(m_critSection);
 
-  for(int msgCount = m_vecThreadMessages.size(); !m_vecThreadMessages.empty() && msgCount > 0; --msgCount)
+  while(!m_vecThreadMessages.empty())
```
Comment (Rechi) [inline xbmc/guilib/GUIWindowManager.cpp]: `while (...` (missing white space)

**Comment by ace20022** [comment]
Forgot one file. Ok, if all lamps go green?

**Review by Rechi — APPROVED** [review]

**Pull request merged**